### PR TITLE
Issue 51 sync with backend axios config (qs, getExperts, getPosts latest-by-direction)

### PR DESCRIPTION
**Pull request by DelFlumen**
develop

## GitHub Board

**Issue link**

[#51  ]( )

**Story link**

## Code reviewers
- [ ] @github_username

## Summary of issue

- [ ]  Summary of issue
- [ ]  Summary of issue

## Summary of change

- [ ]  Summary of change
- [ ]  Summary of change

### CODE REVIEW

File: src/lib/utilities/API/api.ts [hunk 1]
```diff
@@ -12,35 +14,59 @@ const instance = axios.create({
...
   postsType: PostsTypeType,
   config?: GetConfigType,
 ): Promise<AxiosResponse<GetResponseType<PostResponseType>>> => {
-  return instance.get<GetResponseType<PostResponseType>>(
-    `/post/${postsType}`,
-    config,
-  );
+  return instance.get<GetResponseType<PostResponseType>>(`/post/${postsType}`, {
+    ...config,
+    ...defaultConfig,
```
Comment (SergeOlabin) [inline src/lib/utilities/API/api.ts:62]: params config should redefine default, not vice versa. 
```suggestion
     
    ...defaultConfig,
    ...config,
```

File: src/lib/utilities/API/api.ts [hunk 2]
```diff
@@ -12,35 +14,59 @@ const instance = axios.create({
...
 };
 
-type PostsTypeType = 'important' | 'latest';
+export type GeneralGetType =
+  | GetConfigType
+  | GetTagConfigType
+  | GetExpertsConfigType;
+
+const defaultConfig = {
+  paramsSerializer: (params: GeneralGetType): string => {
```
Comment (SergeOlabin) [inline src/lib/utilities/API/api.ts:49]: don't specify a concrete type here.
`{ [key: string]: any }` should be enough

File: src/lib/utilities/API/api.ts [hunk 3]
```diff
@@ -12,35 +14,59 @@ const instance = axios.create({
...
-type PostTagRequestBodyType = {
+export type PostTagRequestBodyType = {
   tag: string;
 };
 
-type PostsTypeType = 'important' | 'latest';
+export type GeneralGetType =
+  | GetConfigType
+  | GetTagConfigType
+  | GetExpertsConfigType;
```
Comment (SergeOlabin) [inline src/lib/utilities/API/api.ts:46]: remove

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/lib/utilities/API/api.ts (hunk 3 above)
Comment (DelFlumen) [reply src/lib/utilities/API/api.ts]: fixed

File: src/lib/utilities/API/api.ts (hunk 2 above)
Comment (DelFlumen) [reply src/lib/utilities/API/api.ts]: fixed

File: src/lib/utilities/API/api.ts (hunk 1 above)
Comment (DelFlumen) [reply src/lib/utilities/API/api.ts:56]: fixed

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**